sdks/python/apache_beam/transforms/core.py

```diff
@@ -2581,7 +2581,7 @@ def add_input_types(transform):
       if pcoll.windowing.windowfn != GlobalWindows():
```

Comment:
We should probably make this stronger, i.e. prohibiting anything but the default (global windows *and* no non-trivial triggering) windowing. IIRC, pcoll.windowing.is_default() would solve this. I don't think the solution below would give the expected output(emitting exactly once towards the beginning of a pipeline the few values that were combined up to that point).

Reply:
I think I understand what you mean now. I changed the condition using what you suggested. But below code has this error `GroupByKey cannot be applied to an unbounded PCollection with global windowing and a default trigger
pipeline_result = pipeline.run()`:
```
pipeline = beam.Pipeline(DataflowRunner(), options=options)
_ = (
    pipeline
    | beam.io.ReadFromPubSub(topic=topic)
    | beam.WindowInto(
        window.GlobalWindows(),
        trigger=trigger.Repeatedly(trigger.AfterCount(4)),
        accumulation_mode=trigger.AccumulationMode.DISCARDING,
        #accumulation_mode=trigger.AccumulationMode.ACCUMULATING,
    )
    | "Print 1" >> beam.transforms.util.LogElements()
    # | beam.combiners.Count.Globally().without_defaults()
    | beam.combiners.Count.Globally()
    | "Print 2" >> beam.transforms.util.LogElements()
)
```

Is it possible to use this combination (Unbounded souce + GlobalWindows with DefaultTrigger)?